### ISSUE TITLE
getValue()  bad update when deep dependencies  

### ISSUE THREAD

**Issue by RelicOfTesla** [json-editor/json-editor#1364]
### General information

- json-editor version: 2.10.0

### Expected behavior
```
{
  "engine": "none"
}
```

### Actual behavior
```
{
  "engine": "none",
  "child1": {
    "L2Name": "aaa"
  }
}
```

### Steps to reproduce the behavior
1. select engine "test1"
2. input "L2Name"="aaa", input "L1Name"="bbb"
3. save / getValue() / get output  (https://pmk65.github.io/jedemov2/dist/demo.html)
4. got good value. back editor
5.  select engine "none"
6.  save / getValue() / get output  (https://pmk65.github.io/jedemov2/dist/demo.html)
7.   "L1Name" is good, don't get last field value "L2Name"

Direct link to example: https://is.gd/MHRH9Q

```javascript
{
	"type": "object",

	"properties": {
		"engine": {
			"type": "string",
			"enum": ["none", "test1"]
		},
		"child1": {
			"type": "object",
			"properties": {
				"L2Name": {
					"type": "string",
					"options": {
						"dependencies": {
							"root.engine": "test1"
						}
					}
				}
			}
		},
		"L1Name": {
			"type": "string",
			"options": {
				"dependencies": {
					"root.engine": "test1"
				}
			}
		}
	}
}
```

**Comment by schmunk42** [comment]
Initial output is:

```
{
  "engine": "none",
  "child1": {
    "L2Name": ""
  },
  "L1Name": ""
}
```

Is this expected with dependencies, despite `Object properties required by default`?

or should it also be:

```
{
  "engine": "none"
}
```

**Comment by germanbisurgi** [comment]
Fixed in #1365
